src/lib/components/Widget.js

```diff
@@ -0,0 +1,262 @@
+              author {
+                login
+              }
+              files(first: 10) {
+                nodes {
+                  path
+                }
+              }
+              mergedAt
+            }
+          }
+        }
+      }`,
+      }),
+    }).then((res) => res.json());
+    pullRequests.data.search.nodes.sort((a,b) => new Date(b.mergedAt) - new Date(a.mergedAt))
+    return pullRequests;
+  }
+
+  async fetchFileContents(list) {
+    var returnList = [];
+    var promises = [];
+    for (var i = 0; i < this.state.numberOfRules; i++) {
+      promises.push(
+        fetch("https://api.github.com/graphql", {
+          method: "POST",
+          headers: {
+            Authorization: "bearer /* INSERT TOKEN */",
+          },
+          body: JSON.stringify({
+            query: `{
+          rateLimit {
+            remaining
+          }
+          repository(name: "SSW.Rules.Content", owner: "SSWConsulting") {
+            object(expression: "main:${list[i]}") {
+              ... on Blob {
+                text
+              }
+            }
+          }
+        }`,
+          }),
+        })
+      );
+    }
+    await Promise.all(promises)
+      .then((values) => Promise.all(values.map((res) => res.json())))
+      .then((values) =>
+        values.forEach((obj) => {
+          if (obj.data.repository.object != null) {
+            returnList = [...returnList, obj.data.repository.object.text];
+          } else {
+            returnList = [...returnList, null];
+          }
+        })
+      );
+    if (
+      returnList.filter((x) => x !== null).length < this.state.numberOfRules
+    ) {
+      var counter = 0;
+      while (
+        returnList.filter((x) => x !== null).length < this.state.numberOfRules
+      ) {
+        var response = await fetch("https://api.github.com/graphql", {
+          method: "POST",
+          headers: {
+            Authorization: "bearer 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",
+          },
+          body: JSON.stringify({
+            query: `{
+            rateLimit {
+              remaining
+            }
+            repository(name: "SSW.Rules.Content", owner: "SSWConsulting") {
+              object(expression: "main:${
+                list[counter + this.state.numberOfRules]
+              }") {
+                ... on Blob {
+                  text
+                }
+              }
+            }
+          }`,
+          }),
+        }).then((res) => res.json());
+        if (response.data.repository.object != null) {
+          returnList = [...returnList, response.data.repository.object.text];
+        } else {
+          returnList = [...returnList, null];
+        }
+        counter++;
+      }
+    }
+    return returnList;
+  }
+
+  async componentDidMount() {
+    this.determineTheme();
+    const pullRequests = await this.fetchPullRequests();
+    var retrievalList = [];
+    var rulesList = [];
+    for (let pr of pullRequests.data.search.nodes) {
```

Comment:
It should be easier to tell what this is doing at a glance 
e.g. move into separate function